#CMake Error string sub-command REPLACE requires at least four arguments.
diff --git a/CMakeLists.txt b/CMakeLists.txt
index ff147e2e..ff878291 100644
--- a/auxil/spicy/spicy/CMakeLists.txt
+++ b/auxil/spicy/spicy/CMakeLists.txt
@@ -26,7 +26,7 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 # CMake uses -O2 by default with RelWithDebInfo.
-string(REPLACE "-O2" "-O3" CMAKE_CXX_FLAGS_RELWITHDEBINFO ${CMAKE_CXX_FLAGS_RELWITHDEBINFO})
+string(REPLACE "-O2" "-O3" CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO}")
 
 include(CheckCompiler)
 
diff --git a/CMakeLists.txt b/CMakeLists.txt
index f8f7a70..8b04bb4 100644
--- a/auxil/spicy-plugin/CMakeLists.txt
+++ b/auxil/spicy-plugin/CMakeLists.txt
@@ -75,7 +75,7 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 # CMake uses -O2 by default with RelWithDebInfo.
-string(REPLACE "-O2" "-O3" CMAKE_CXX_FLAGS_RELWITHDEBINFO ${CMAKE_CXX_FLAGS_RELWITHDEBINFO})
+string(REPLACE "-O2" "-O3" CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO}")
 
 include_directories(BEFORE ${PROJECT_BINARY_DIR}/include)
 include_directories(BEFORE ${PROJECT_SOURCE_DIR}/include)
